tests/cxx/no_def_arg_call_site.cc

```diff
@@ -0,0 +1,30 @@
+//===--- no_def_arg_call_site.cc - test input file for iwyu ---------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// IWYU_ARGS: -I .
+
+// Tests that IWYU doesn't report default function arguments when analyzing call
+// site.
+
+#include "tests/cxx/no_def_arg_call_site-d1.h"
+
+template <typename T>
+void TplFn() {
+  T::FnWithDefArg();
+}
+
+void Fn() {
+  TplFn<Struct>();
+}
```

Comment:
Would it be prudent to test `Struct::FnWithDefArg();` directly here? Just to have coverage outside of template too.

Reply:
Without template context, symbols used inside default argument expression are correctly reported to the function declaration location, so there was no observable bug, just redundancy in traversal. `InstantiatedTemplateVisitor` reports them to the instantiation location.

Reply:
Yeah, I mean it would be nice to show that here, for completeness.